src/tool_operate.c

```diff
@@ -888,6 +892,93 @@ static CURLcode single_transfer(struct GlobalConfig *global,
           }
         }
 
+        /* disallowing simultaneous use of --etag-save and --etag-compare */
+        if(config->etag_save_file && config->etag_compare_file) {
+          warnf(
+            config->global,
+            "Cannot use --etag-save and --etag-compare at the same time\n");
+
+          result = CURLE_UNKNOWN_OPTION;
+          break;
+        }
+
+        /* --etag-save */
+        etag_save = &per->etag_save;
+        etag_save->stream = stdout;
+        etag_save->config = config;
+        if(config->etag_save_file) {
+          /* open file for output: */
+          if(strcmp(config->etag_save_file, "-")) {
+            FILE *newfile = fopen(config->etag_save_file, "wb");
+            if(!newfile) {
+              warnf(
+                config->global,
+                "Failed to open %s\n", config->etag_save_file);
+
+              result = CURLE_WRITE_ERROR;
+              break;
+            }
+            else {
+              etag_save->filename = config->etag_save_file;
+              etag_save->s_isreg = TRUE;
+              etag_save->fopened = TRUE;
+              etag_save->stream = newfile;
+            }
+          }
+          else {
+            /* always use binary mode for protocol header output */
+            set_binmode(etag_save->stream);
+          }
+        }
+
+        /* --etag-compare */
+        if(config->etag_compare_file) {
+          char *etag_from_file;
+          size_t lgn;
+          char *header;
+
+          /* open file for reading: */
+          FILE *file = fopen(config->etag_compare_file, FOPEN_READTEXT);
+          if(!file) {
+            warnf(
+              config->global,
+              "Failed to open %s\n", config->etag_compare_file);
+
+            result = CURLE_READ_ERROR;
+            break;
+          }
+
+          /* get etag from file as string, and a length of it */
+          file2memory(&etag_from_file, &lgn, file);
+
+          if(etag_from_file == NULL) {
+            warnf(
+              config->global,
+              "File: %s is empty or in wrong format\n",
+              config->etag_compare_file);
+
+            result = CURLE_READ_ERROR;
+            break;
+          }
+
+          /* remove trailing new line from etag */
+          etag_from_file[strcspn(etag_from_file, "\n")] = 0;
+
+          /*
+           * create buffer to store constructed custom header
+           * 18 because string If-None-Match:
+           * (+ quotation marks and line terminator) is 18 characters
+           */
+          header = malloc(18 + lgn);
```

Comment:
This should check for a NULL return and if so bail out.

Reply:
done